tests/functional_tests/util_resources.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 The Monero Project
+# 
+# All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+# 
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+# 
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+# 
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+    Help determine how much CPU power is available at the given time
+    by running numerical calculations
+"""
+
+from __future__ import print_function
```

Comment:
Makes no sense in python3.

Reply:
Until https://github.com/monero-project/monero/pull/7396 is merged, we're stuck with Python2

Reply:
I see your point, but I think, perhaps, you've ended up making your life more difficult by splitting commits in this way: Though each commit and PR should achieve a single logical goal, there are cases where a clear dependency exists, and ignoring the dependency therefore introduces extra work and complexity, the very reasons to ordinarily avoid PRs with multiple aims. For example, this PR adds multiple python2-specific pieces of code, which will eventually have to be removed. On top of that, the files all still mark themselves as targeting python3 via the shebang line, which makes life very confusing for the reader. You could have, instead, either branched this PR from the "use python3" PR, or introduced a more limited change which just runs the mining tests with python3, as a single commit which forms part of this multi-commit PR. I don't think it's an issue worthy of further discussion, but certainly something to consider in future work.